File: package/umt_python/src/object/path_segments.py
def path_segments(path: str | list[str]) -> list[str]:
    """
    Normalizes a property path into a list of segments.
    A dotted string is split on ".", while a list is returned unchanged.

    A list input is returned as-is (the same object, not a copy), mirroring the
    TypeScript `Object/pathSegments` behavior.

    Args:
        path (str | list[str]): Dot-separated string or list of path segments.

    Returns:
        list[str]: The path split into segments.

    Example:
        >>> path_segments("a.b.c")
        ['a', 'b', 'c']
        >>> path_segments(["a", "b"])
        ['a', 'b']
        >>> path_segments("a")
        ['a']
        >>> path_segments("")
        ['']
    """
    return path.split(".") if isinstance(path, str) else path

File: package/umt_rust/src/object/path_segments.rs
/// Splits a dot-separated path string into its segments.
///
/// Mirrors the TypeScript `pathSegments` helper used by `get`, `set` and
/// `unflattenObject`. A string is split on `.`; an empty string yields a
/// single empty segment (matching JavaScript's `"".split(".")`).
///
/// # Arguments
///
/// * `path` - The dot-separated path string.
///
/// # Returns
///
/// A vector of path segments.
///
/// # Examples
///
/// ```
/// use umt_rust::object::umt_path_segments;
///
/// assert_eq!(umt_path_segments("a.b.c"), vec!["a", "b", "c"]);
/// assert_eq!(umt_path_segments("a"), vec!["a"]);
/// assert_eq!(umt_path_segments(""), vec![""]);
/// ```
pub fn umt_path_segments(path: &str) -> Vec<String> {
    path.split('.').map(|segment| segment.to_string()).collect()
}
